working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
+        The term `augmented` has no meaning, and must not occur in
+        optional parameter default value expressions or in
+        target constructor arguments expressions.
+    *   **Redirecting factory constructor**: Contains no expressions.
+
+    The term <code>'augmented' \<argumentPart\></code> words parses
+    as a `<primary>` expression.
 
 In all relevant cases, if the augmented member is an instance member, it is
-invoked with the same value for `this`.
-
-Assume that the identifier `augmented` occurs in a source location where no
-enclosing declaration is augmenting. In this case, the identifier is taken
-to be a reference to a declaration which is in scope.
-
-*In other words, `augmented` is just a normal identifier when it occurs
-anywhere other than inside an augmenting declaration.*
-
-*Note that, for example, `augmented()` is an invocation of the augmented
-function or method when it occurs in an augmenting function or method
-declaration. (In the latter case, the augmenting method declaration must
-occur inside an augmenting type-introducing declaration, e.g., an
-augmenting class or mixin declaration). This is also true if `augmented()`
-occurs inside a local function declaration inside the body of that function
-or method declaration. We could say that `augmented` is a contextual
-reserved word because it is unable to refer to a declaration in scope when
-it occurs inside an augmenting declaration, it always has the special
-meaning which is associated with augmentations.*
-
-A compile-time error occurs if a declaration with the basename `augmented`
-occurs in a location where any enclosing declaration is augmenting. *This
-error is applicable to all such declarations, e.g., local functions, local
-variables, parameters, and type parameters.*
-
-Consider a non-augmenting member declaration _Dm_ that occurs inside an
-augmenting type declaration _Dt_. A compile-time error occurs if the
-identifier `augmented` occurs in _Dm_.
-
-*For example, inside `augment class C` we could have a declaration like
-`void f() {...augmented()...}`. This is an error because the outer
-`augment` forces the meaning of `augmented` to be about augmentation in the
-entire scope, but the method declaration is introductory and hence there is
-no earlier declaration to augment.*
+invoked with the same value for `this`. For a non-redirecting factory
+constructor, it's invoked with the same type parameter bindings for the
+surrounding type.
+
+The grammar change for `augmented` adds the following productions:
+```ebnf
+  <augmentedValueExpression> ::= 'augmented'
+  <augmentedSetterExpression> ::= 'augmented' '=' <expression>
+  <augmentedFunctionExpression> ::= 'augmented' <argumentPart>
+  <augmentedConstructorExpression> ::= 'augmented' <argumentList>
+
+  <primary> ::= ...
+     | <augmentedValueExpression>        -- augmenting variable or getter body
+     | <augmentedSetterExpression>       -- augmenting setter body
```

Comment:
Is slotting this into `<primary>` correct? Wouldn't that allow:

```dart
1 + augmented = 2;
```

Reply:
Good point, it would.
So it should be moved to where assignment expressions live, close to `expression`.

I'm used to thinking of `primary.selector.selector = expr` as a "selector chain", because of how `?.` extends through it, but it's not. It's an assignment expression that has a selector chain as LHS. (And that's also why `++foo?.bar` is allowed. Let's not do that again.)